File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
public class Solution {
 
    public int solution(int[] numbers, int target) {
       return bfs(numbers,0,target,0);
    }
    
    int bfs(int[] numbers, int index, int target, int cur) {
        if(index == numbers.length){
            if(cur == target) {
                return 1;
            }
            return 0;
        }
        
        int sum = 0;
        sum += bfs(numbers,index+1,target,cur+numbers[index]);
        sum += bfs(numbers,index+1,target,cur-numbers[index]);
        
        return sum;
    }
    
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    
    def dfs(index, total) :
        nonlocal answer
        
        # 숫자를 모두 사용한 경우
        if index == len(numbers) :
            if total == target :
                answer += 1
            return
        
        #현재 숫자를 더하는 경우
        
        dfs(index + 1, total + numbers[index])
        
        #현재 숫자를 빼는 경우
        
        dfs(index + 1, total - numbers[index])
        
    dfs(0,0)
    
    
    return answer